### PR TITLE
created initial GUI

### PR DESCRIPTION
**Pull request by harshkpatel**
This is the initial GUI, there is still some refactoring that needs to be done, which I will get to. Tests also need to be written for everything in the frontend with @susannah-c will be responsible for.

### CODE REVIEW

File: src/main/java/UseCases/PresenterDisplay.java [hunk 1]
```diff
@@ -1,55 +1,46 @@
...
-    /**
-     * Function that runs the game loop by getting game data from the IOController and presenting that to the
-     * user as their options on each turn, getting the user's input and sending that back to the IOController
+     * Function that runs the game loop by getting game data from the OutputInteractor and presenting that to the
+     * user as their options on each turn through Conversion in the JDisplay,
+     *  and then getting the user's input and sending that back to the InputInteractor
      * to further handle state changes based on their option choice.
      **/
     public void playGame(File file){
         UseCaseInteractor interactor = new UseCaseInteractor(new LoadFile(file));
```
Comment (susannah-c) [inline src/main/java/UseCases/PresenterDisplay.java:23]: I believe UseCaseInteractor takes in 2 parameters now. There are a few changes made in UseCaseInteractor class.

**Review by susannah-c — APPROVED** [review]
There are some minor fixes that need to be changed. The GUI is also very helpful instead of just playing it on the console. I also think deleting addOptionStrings() and replacing it is smart.

**Review by benxu3 — APPROVED** [review]
Good idea to create a button factory and address the potential switch state code smell by creating the Button classes. Thanks for adding Java Doc to your methods, we will just need to ensure that the tests which Susannah will write will cover these methods. Overall, good work!

File: src/main/java/GUI/ButtonFactory.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    public void addStrings(ArrayList<String> options){
+        this.buttonStrings = options;
+    }
+
+    /**
+     * Function that creates all the ButtonHandlers and adds them to the Arraylist
+     * of ButtonHandlers
+     */
+    public void addButtons(){
+        Button1 b1 = new Button1();
```
Comment (Cindywwang) [inline src/main/java/GUI/ButtonFactory.java:44]: Maybe we could refractor this by extracting a method or using a four loop to create and add 16 buttons

**Review by Cindywwang — APPROVED** [review]

**Pull request merged**